docs/source/cpp/acero/user_guide.rst

```diff
@@ -0,0 +1,735 @@
+provide an asycnhronous stream of batches and is covered in more detail :ref:`here <stream_execution_source_docs>`.
+
+.. _ExecNode List:
+
+Available ``ExecNode`` Implementations
+======================================
+
+The following tables quickly summarize the available operators.
+
+Sources
+-------
+
+These nodes can be used as sources of data
+
+.. list-table:: Source Nodes
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Factory Name
+     - Options
+     - Brief Description
+   * - ``source``
+     - :class:`SourceNodeOptions`
+     - A generic source node that wraps an asynchronous stream of data (:ref:`example <stream_execution_source_docs>`)
+   * - ``table_source``
+     - :class:`TableSourceNodeOptions`
+     - Generates data from an :class:`arrow::Table` (:ref:`example <stream_execution_table_source_docs>`)
+   * - ``record_batch_source``
+     - :class:`RecordBatchSourceNodeOptions`
+     - Generates data from an iterator of :class:`arrow::RecordBatch`
+   * - ``record_batch_reader_source``
+     - :class:`RecordBatchReaderSourceNodeOptions`
+     - Generates data from an :class:`arrow::RecordBatchReader`
+   * - ``exec_batch_source``
+     - :class:`ExecBatchSourceNodeOptions`
+     - Generates data from an iterator of :class:`arrow::compute::ExecBatch`
+   * - ``array_vector_source``
+     - :class:`ArrayVectorSourceNodeOptions`
+     - Generates data from an iterator of vectors of :class:`arrow::Array`
+   * - ``scan``
+     - :class:`arrow::dataset::ScanNodeOptions`
+     - Generates data from an `arrow::dataset::Dataset` (requires the datasets module)
+       (:ref:`example <stream_execution_scan_docs>`)
+
+Compute Nodes
+-------------
+
+These nodes perform computations on data and may transform or reshape the data
+
+.. list-table:: Compute Nodes
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Factory Name
+     - Options
+     - Brief Description
+   * - ``filter``
+     - :class:`FilterNodeOptions`
+     - Removes rows that do not match a given filter expression
+   * - ``project``
+     - :class:`ProjectNodeOptions`
+     - Creates new columns by evaluating compute expressions.  Can also drop and reorder columns
+       (:ref:`example <stream_execution_project_docs>`)
+   * - ``aggregate``
+     - :class:`AggregateNodeOptions`
+     - Calculates summary statistics across the entire input stream or on groups of data
+       (:ref:`example <stream_execution_aggregate_docs>`)
+   * - ``pivot_longer``
+     - :class:`PivotLongerNodeOptions`
+     - Reshapes data by converting some columns into additional rows
+
+Arrangement Nodes
+-----------------
+
+These nodes reorder, combine, or slice streams of data
+
+.. list-table:: Arrangement Nodes
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Factory Name
+     - Options
+     - Brief Description
+   * - ``hash_join``
+     - :class:`HashJoinNodeOptions`
+     - Joins two inputs based on common columns (:ref:`example <stream_execution_hashjoin_docs>`)
```

Comment:
AsOfJoin is missing here?

Reply:
I added asofjoin to the table.  Regrettably we don't have an example yet but I'll defer that for follow-up.